browser/src/UI/Reducer.ts

```diff
@@ -149,22 +149,27 @@ export const buffersReducer = (s: State.IBufferState, a: Actions.SimpleAction):
         case "BUFFER_ENTER":
             byId = {
                 ...s.byId,
-                [a.payload.id]: {
-                    id: a.payload.id,
-                    file: a.payload.file,
-                    totalLines: a.payload.totalLines,
-                    hidden: a.payload.hidden,
-                    listed: a.payload.listed,
+                [a.payload.currentBuffer.id]: {
```

Comment:
Following from the above comment in the ActionCreator - if we used a single array, as opposed to differentiating the cases, it seems like the lower `reduce` would handle both the current buffer + active buffer cases.